cloud/organization/organization_token.go

```diff
@@ -0,0 +1,190 @@
+)
+
+const (
+	workspaceConst = "WORKSPACE"
+)
+
+func newTokenSelectionTableOut() *printutil.Table {
+	return &printutil.Table{
+		Padding:        []int{44, 50},
+		DynamicPadding: true,
+		Header:         []string{"#", "NAME", "DESCRIPTION"},
+	}
+}
+
+// Update a workspace token
+func AddOrgTokenToWorkspace(id, name, role, workspace string, out io.Writer, client astrocore.CoreClient) error {
+	err := user.IsWorkspaceRoleValid(role)
+	if err != nil {
+		return err
+	}
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	if ctx.OrganizationShortName == "" {
+		return user.ErrNoShortName
+	}
+	if workspace == "" {
+		workspace = ctx.Workspace
+	}
+	tokens, err := getOrganizationTokens(client)
+	if err != nil {
+		return err
+	}
+	token, err := getOrganizationToken(id, name, "\nPlease select the organization token you would like to add to the workspace:", tokens)
+	if err != nil {
+		return err
+	}
+	apiTokenID := token.Id
+
+	var orgRole string
+	for i := range token.Roles {
+		if token.Roles[i].EntityId == workspaceConst {
+			return errOrgTokenInWorkspace
+		}
+
+		if token.Roles[i].EntityId == ctx.Organization {
+			orgRole = token.Roles[i].Role
+		}
+	}
+
+	apiTokenWorkspaceRole := astrocore.ApiTokenWorkspaceRole{
+		EntityId: workspace,
+		Role:     role,
+	}
+	apiTokenWorkspaceRoles := []astrocore.ApiTokenWorkspaceRole{apiTokenWorkspaceRole}
+
+	updateOrganizationAPITokenRoles := astrocore.UpdateOrganizationApiTokenRoles{
+		Organization: orgRole,
+		Workspace:    &apiTokenWorkspaceRoles,
+	}
+	updateOrganizationAPITokenRequest := astrocore.UpdateOrganizationApiTokenRequest{
+		Name:        token.Name,
+		Description: token.Description,
+		Roles:       updateOrganizationAPITokenRoles,
+	}
+
+	resp, err := client.UpdateOrganizationApiTokenWithResponse(httpContext.Background(), ctx.OrganizationShortName, apiTokenID, updateOrganizationAPITokenRequest)
+	if err != nil {
+		return err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "Astro Organization Token %s was successfully added to the Workspace\n", token.Name)
+	return nil
+}
+
+func selectTokens(apiTokens []astrocore.ApiToken) (astrocore.ApiToken, error) {
+	apiTokensMap := map[string]astrocore.ApiToken{}
+	tab := newTokenSelectionTableOut()
+	for i := range apiTokens {
+		name := apiTokens[i].Name
+		description := apiTokens[i].Description
+
+		index := i + 1
+		tab.AddRow([]string{
+			strconv.Itoa(index),
+			name,
+			description,
+		}, false)
+		apiTokensMap[strconv.Itoa(index)] = apiTokens[i]
+	}
+
+	tab.Print(os.Stdout)
+	choice := input.Text("\n> ")
+	selected, ok := apiTokensMap[choice]
+	if !ok {
+		return astrocore.ApiToken{}, errInvalidOrganizationTokenKey
+	}
+	return selected, nil
+}
+
+func getOrganizationToken(id, name, message string, tokens []astrocore.ApiToken) (token astrocore.ApiToken, err error) { //nolint:gocognit
```

Comment:
Why do we need the message as input argument?

Reply:
because the message changes based on what you are doing to to the org token